p2p/peer.py

```diff
@@ -190,6 +190,7 @@ def __init__(self,
         self._subscribers: List[PeerSubscriber] = []
         self.start_time = datetime.datetime.now()
         self.received_msgs: Dict[protocol.Command, int] = collections.defaultdict(int)
+        self.booted = asyncio.Event()
```

Comment:
As I mentioned in https://github.com/ethereum/py-evm/issues/1234#issuecomment-418798308 and https://github.com/ethereum/py-evm/pull/1258#issuecomment-419061543, I'm not super keen on this approach, but if you think it's ok we should at least have docstrings making it clear callsites have to wait for `self.booted` before using the peer. And ideally that should be moved into `Peer.events`, no?

Reply:
Will get this updated.  I'm not inclined to add it to the `Peer.events` yet because that requires a decent bit of boilerplate to get the type checks right *or* to just push the event onto the object at runtime and I don't love either of those yet.  (also, it muddles the Peer API which is special with the Service API which is generic).